src/services/itemLogin/schemas.ts

```diff
@@ -9,89 +10,100 @@ import { errorSchemaRef } from '../../schemas/global';
 import { accountSchemaRef } from '../account/schemas';
 import { itemSchemaRef } from '../item/schemas';
 
-export const credentialsSchemaRef = registerSchemaAsRef(
-  'itemLoginCredentials',
-  'Item Login Credentials',
-  Type.Object(
-    {
-      username: customType.Username(),
-      password: Type.Optional(
-        Type.String({ minLength: 3, maxLength: 50, pattern: '^\\S+( \\S+)*$' }),
-      ),
-    },
-    { additionalProperties: false },
-  ),
-);
-
-const itemLoginSchemaSchema = Type.Object(
+const itemLoginSchemaSchema = customType.StrictObject(
   {
     id: customType.UUID(),
-    type: customType.EnumString(Object.values(ItemLoginSchemaType)),
-    status: customType.EnumString(Object.values(ItemLoginSchemaStatus)),
+    type: customType.EnumString(Object.values(ItemLoginSchemaType), {
+      description: 'Defined which credentials are necessary to login.',
+    }),
+    status: customType.EnumString(Object.values(ItemLoginSchemaStatus), {
+      description:
+        'Whether the item login is enabled, freezed or disabled. An item login is never deleted, but disabled instead to prevent attached guests to be deleted.',
+    }),
     item: Type.Optional(itemSchemaRef),
     createdAt: customType.DateTime(),
     updatedAt: customType.DateTime(),
   },
-  { additionalProperties: false },
+  {
+    description:
+      'Instance allowing to login without a member on related item and its descendants. The required credentials are defined given the status.',
+  },
 );
+
 export const itemLoginSchemaSchemaRef = registerSchemaAsRef(
   'itemLoginSchema',
   'Item Login Schema',
   itemLoginSchemaSchema,
 );
 
-export const login = {
+export const loginOrRegister = {
+  operationId: 'itemLogin',
+  tags: ['item-login'],
+  summary: 'Login or Register in item',
+  description: `Login in item with necessary credentials depending on item login's status. If the username does not exist, a guest account is created and the user is logged in.`,
+
   params: customType.StrictObject({
     id: customType.UUID(),
   }),
-  body: credentialsSchemaRef,
+  body: customType.StrictObject({
+    username: customType.Username(),
+    password: Type.Optional(
+      Type.String({ minLength: 3, maxLength: 50, pattern: '^\\S+( \\S+)*$' }),
```

Comment:
Should we extract this pattern to name it and simplify the readability? 
```suggestion
      Type.String({ minLength: 3, maxLength: 50, pattern: 'SingleSpacedWords' }),
```

Reply:
Is this regex really useful ? what guaranties does it give us ? can't the password simply be a string ?